src/ralph/lib/permissions/models.py

```diff
@@ -12,31 +12,133 @@
 
 
+def get_perm_key(action, class_name, field_name):
+    """
+    Generated django permission code name.
+
+    :Example:
+
+        >> perm = get_perm_key('change', 'baseobject', 'remarks')
+        >> print(perm)
+        >> change_baseobject_remarks_field
+
+    :param action: Permission action (change/view)
+    :param class_name: Django model class name
+    :param field_name: Django model field name
+
+    :type action: str
+    :type class_name: str
+    :type field_name: str
+
+    :return: django permission code name
+    :rtype: str
+    """
+    return '{}_{}_{}_field'.format(action, class_name, field_name)
+
+
 class PermissionByFieldBase(ModelBase):
-    """Extends permissions by edit per field of model."""
+
+    """
+    Metaclass added django permissions based on all fields in the model.
+
+    :Example:
+
+        class Test(with_metaclass(PermissionByFieldBase, models.Model)):
+
+            ...
+
+            class Permissions:
+                # Fields to exclude generated permissions
+                blacklist = set(['service_env'])
+    """
+
     def __new__(cls, name, bases, attrs):
         new_class = super(PermissionByFieldBase, cls).__new__(
             cls, name, bases, attrs
         )
+        class_name = new_class._meta.model_name
         for field in new_class._meta.fields:
+            blacklist = new_class.Permissions.blacklist
             name = field.name
-            new_class._meta.permissions.append((
-                'can_edit_{}_field'.format(name),
-                _('Can edit {} field').format(name)
-            ))
+            if not field.primary_key and name not in blacklist:
+                new_class._meta.permissions.append((
+                    get_perm_key('change', class_name, name),
+                    _('Can change {} field').format(field.verbose_name)
+                ))
+                new_class._meta.permissions.append((
+                    get_perm_key('view', class_name, name),
+                    _('Can view {} field').format(field.verbose_name)
+                ))
+
         return new_class
 
 
 class PermByFieldMixin(with_metaclass(PermissionByFieldBase, models.Model)):
 
-    # TODO
-    def have_access_to_field(self, field, user=None, group=None):
-        pass
+    """Django Abstract model class for permission by fields."""
+
+    def has_access_to_field(self, field_name, user, action='change'):
+        """
+        Checks the user has the permission to the field
 
-    # TODO
-    @property
-    def allowed_fields(self, user=None, group=None):
-        pass
+        :Example:
+
+            >> user = User.objects.get(username='root')
+            >> print(model.has_access_to_field('remarks', user, action='change'))
+            >> True
+
+        :param field_name: django model field name
+        :param user: django User object
+        :param action: permission action (change/view)
+
+        :type field_name: str
```

Comment:
to discuss if we want first definition of all fields and then all types or mixed (field, type, field, type, ...). @ar4s @vi4m @xliiv what do you think?